### ISSUE TITLE
date-time format error. 

### ISSUE THREAD

**Issue by jcoyne** [byu-oit/openapi-enforcer#30]
I'm getting a warning 
```
Document is valid but has one or more warnings. One or more warnings exist in the OpenApi definition at: components > schemas > Access > properties > embargoReleaseDate > example Unable to deserialize value Expected a date-time string of the format YYYY-MM-DDTmm:hh:ss.sssZ
```

about this example:
https://github.com/sul-dlss/dor-services-app/blob/ede5786209b60577b5c47d3b6452ccb041f72e59/openapi.json#L899

My example date is `""2029-06-22T07:00:00.000+00:00"` and I believe this is an rfc3339 compliant date as https://tools.ietf.org/html/rfc3339#section-5.6 states:

```
   time-numoffset  = ("+" / "-") time-hour ":" time-minute
   time-offset     = "Z" / time-numoffset
```

 See examples at: https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#date.and.time.format.examples

**Comment by Gi60s** [comment]
You're right. Thanks for the catch. I'll get working on a fix.

**Comment by Gi60s** [comment]
Fixed with PR #32 and published to NPM as version 1.5.2.

**Comment by jcoyne** [comment]
Thank you @Gi60s!

**Comment by delanym** [comment]
I get an error with date-time value "2020-12-13T12:02:33+0000" without a colon in the offset. This is also valid RFC3339

**Comment by Gi60s** [comment]
Interesting. I don't see where that is in the spec but JavaScript does recognize it. Can you point me more specifically to where the spec shows it, maybe even quote it?

**Comment by delanym** [comment]
https://github.com/OAI/OpenAPI-Specification/issues/2374

**Comment by Gi60s** [comment]
I guess what I'm asking for is where in RFC3339 does it mention that an offset can omit the `:` (colon)? Here is what I see at https://tools.ietf.org/html/rfc3339#section-5.6:

```
The following profile of ISO 8601 [ISO8601] dates SHOULD be used in
new protocols on the Internet.  This is specified using the syntax
description notation defined in [ABNF].

date-fullyear   = 4DIGIT
date-month      = 2DIGIT  ; 01-12
date-mday       = 2DIGIT  ; 01-28, 01-29, 01-30, 01-31 based on month/year
time-hour       = 2DIGIT  ; 00-23
time-minute     = 2DIGIT  ; 00-59
time-second     = 2DIGIT  ; 00-58, 00-59, 00-60 based on leap second rules
time-secfrac    = "." 1*DIGIT
time-numoffset  = ("+" / "-") time-hour ":" time-minute
time-offset     = "Z" / time-numoffset

partial-time    = time-hour ":" time-minute ":" time-second[time-secfrac]
full-date       = date-fullyear "-" date-month "-" date-mday
full-time       = partial-time time-offset

date-time       = full-date "T" full-time
```

The `time-numoffset` shows a colon. What am I missing where it says that the colon is optional?

**Comment by delanym** [comment]
Its in appendix A

> time-numoffset    = ("+" / "-") time-hour [[":"] time-minute]

There's another issue. The spec says in section 5.6

> NOTE: ISO 8601 defines date and time separated by "T". Applications using this syntax may choose, for the sake of readability, to specify a full-date and full-time separated by (say) a space character.

But then in appendix A it contradicts that

> ISO 8601 states that the "T" may be omitted under some circumstances. This grammar requires the "T" to avoid ambiguity.

**Comment by Gi60s** [comment]
Who knew dates were so complex? Only a few of us I suppose. Thanks for the reference.

I've published a fix to NPM as version `1.11.1`.

**Comment by delanym** [comment]
On my way to the time, I met a date with seven specs. Each spec (with seven grammars) had seven profiles and seven types. Each spec had seven sections, each spec with several versions. Each spec deferred to the next, and over time had seven errata. Each spec—seven validators, in seven languages, each failure seven tickets, each with seven comments, and seven hundred views. Specs, sections, versions, profiles, ratas, types, tickets, gramms, langs, comments and views: how many date-times did I need to know which position to put year/month/day?
